cpp/src/io/laz_writer.cpp

```diff
@@ -0,0 +1,44 @@
+#include "copc-lib/io/laz_writer.hpp"
+
+#include <memory>
+
+namespace copc::laz
+{
+
+LazWriter::LazWriter(std::ostream &out_stream, const las::LazConfigWriter &laz_config_writer)
+    : BaseWriter(out_stream,
+                 std::static_pointer_cast<las::LazConfig>(std::make_shared<las::LazConfigWriter>(laz_config_writer)))
```

Comment:
can we move this to the constructor body, so we don't have logic in this section?

Reply:
Not sure how to do that?